### PR TITLE
issue-1-create-an-entry - Skylar Shaffer

### PR DESCRIPTION
**Pull request by skylarshaffer**
https://github.com/skylarshaffer/code-journal/assets/161654841/e3c683d1-f9b3-4c3a-a38d-c194eac1cbcb

### CODE REVIEW

File: ts/data.ts [hunk 1]
```diff
@@ -1,8 +1,24 @@
...
+  title: string;
+  photoUrl: string;
+  notes: string;
+  entryId: number;
+}
 
 let data = {
   view: 'entry-form',
-  entries: [],
+  entries: [] as Entry[],
```
Comment (shawnkost) [inline ts/data.ts:11]: Rather than doing `as` just do `entries: Entry[]`

File: ts/main.ts [hunk 2]
```diff
@@ -1 +1,42 @@
...
+const $formImg = document.querySelector('#formImg') as HTMLImageElement;
+const $form = document.querySelector('form') as HTMLFormElement;
+
+//  error coverage
+if (!$photoUrl || !$formImg || !$form)
+  throw new Error('One of the dom queries failed');
+
+//  $photoUrl handleInput
+$photoUrl.addEventListener('input', (event: Event) => {
+  const eventTarget = event.target as HTMLInputElement | HTMLTextAreaElement;
```
Comment (shawnkost) [inline ts/main.ts:20]: This should just be `HTMLInputElement`. The event is only on `$photoUrl` which is a `HTMLInputElement`

**Review by shawnkost — CHANGES_REQUESTED** [review]
Looks great Skylar! Just a couple small tweaks requested

File: ts/data.ts (hunk 1 above)
Comment (skylarshaffer) [reply ts/data.ts:11]: I think this would only work in an interface, right?
image

File: ts/main.ts (hunk 2 above)
Comment (skylarshaffer) [reply ts/main.ts:20]: Thanks! That was an artifact from typescript-forms, I believe. Fixing.

File: ts/data.ts (hunk 1 above)
Comment (skylarshaffer) [reply ts/data.ts]: Fixed
```JavaScript
interface Data {
  view: string;
  entries: Entry[];
  editing: null;
  nextEntryId: number;
}

let data: Data = {
  view: 'entry-form',
  entries: [],
  editing: null,
  nextEntryId: 1,
};
```

**Review by shawnkost — APPROVED** [review]
👍🏻

**Pull request merged**